### PR TITLE
[AWS] Add support for Inspector datastream and Remove duplicate ECS fields

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Added a data stream.
- Added data collection logic to the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files
- Added dashboards and visualizations.
- Added test for pipeline for the data stream.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.4.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/aws/data_stream directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Related https://github.com/elastic/integrations/issues/4398

## Screenshots

image
image
image
image
image
image
image

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
@vinit-elastic Could we actually remove the duplicate fields from agent.yml and keep them in the ecs.yml instead? Thanks!! I just did this for the ec2 integration: https://github.com/elastic/integrations/pull/4567

**Comment by vinit-elastic** [comment]
@kaiyan-sheng the reason why I removed fields from ecs.yml is because the agent.yml is a static file and is the same across all the connectors. And the same has been discussed with @P1llus earlier, therefore we didn't change it.

**Comment by P1llus** [comment]
@vinit-elastic I agree, if we want to change agent.yml we should simply do it in one go for all packages instead @kaiyan-sheng ?

**Review by andrewctate — CHANGES_REQUESTED** [review]
Hi @vinit-elastic! I'm going to request that we use the new Lens metric vis instead of the TSVB metric on the dashboard since it's a good fit and we're trying to get away from TSVB. Here's a screenshot of how you can configure it in the Lens editor:

Screen Shot 2022-11-10 at 3 27 39 PM

And a couple "softer" requests: 

1. Could you post a screenshot of this dashboard with data in it?

2. This is a big dashboard (16 visualizations). The concern with these is that they can load really slowly if the customer has a lot of data. All visualizations (including tables) (re)render every time the user navigates to the dashboard/adds a filter/changes the time range, regardless of whether or not they are above the fold. Is there a logical way it could make sense to break this up into smaller dashboards connected by hyperlinks?

**Review by andrewctate — APPROVED** [review]
Thanks for making the visualization type change!

**Comment by kaiyan-sheng** [comment]
@vinit-elastic How about lets keep the duplicated ECS fields for now, and make a separate PR to remove all the duplicate fields from agent.yml instead?

**Comment by vinit-elastic** [comment]
> @vinit-elastic How about lets keep the duplicated ECS fields for now, and make a separate PR to remove all the duplicate fields from agent.yml instead?

Hey @kaiyan-sheng, if we keep the duplicate fields as they are, then the elastic-package check command gives an error. because both files have different types of descriptions and there is a type change in the newer Stack version as well as the ECS version. The same has been discussed with @P1llus earlier and we came to a conclusion that keep agent.yml as it is and remove duplicate fields in the ECS file.

**Comment by kaiyan-sheng** [comment]
@vinit-elastic OK! Thanks for the explanation. I will make a separate PR later after this one gets in to fix the agent.yml and ecs.yml then!

File: packages/aws/manifest.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 format_version: 1.0.0
 name: aws
 title: AWS
-version: 1.25.2
+version: 1.27.0
```
Comment (kaiyan-sheng) [inline packages/aws/manifest.yml:4]: Should we back this one to 1.26.0 for now?

**Comment by kaiyan-sheng** [comment]
I made the change for removing duplicate fields from agent.yml and use ecs.yml instead in PR: https://github.com/elastic/integrations/pull/4657

**Comment by vinit-elastic** [comment]
Hey @kaiyan-sheng - I saw your PR, appreciate your effort. 
However, we are doing the opposite in all the connectors. That is, we keep the agent.yml file same and remove the duplicates from ecs.yml. so that agent.yml is the same across all the integrations. So could you please do the same? Thanks!

**Comment by kaiyan-sheng** [comment]
@vinit-elastic @P1llus Sorry thats the part I don't quite get. Why we are redefining ECS fields in `agent.yml` files? For example `cloud.availability_zone`. Unless we are using `cloud.availability_zone` in a different way than what ECS field defines, we should keep it in `ecs.yml` to leverage ECS as much as we can.

**Comment by P1llus** [comment]
@kaiyan-sheng Its mostly because all the integrations have been setup like this as far as I know, the fields are simply copied from the other integrations.

We could change it, but I rather keep it out of this PR and do it over all integrations if you want. There is plenty of ECS fields in agent.yml from before.
I am unsure what the difference is, its still an ECS field, we are not redefining it, its simply in another file.

**Review by kaiyan-sheng — APPROVED** [review]
@P1llus and I had a conversation offline and decides to merge my PR for moving all fields into ecs.yml first to resolve this issue. Sorry @vinit-elastic you have to rebase your PR and sorry for blocking it for a while... The part for adding support for inspector data stream looks good to me. Thank you both!!

File: packages/aws/manifest.yml (hunk 1 above)
Comment (vinit-elastic) [reply packages/aws/manifest.yml:4]: Hey @kaiyan-sheng, I have raised another PR related to [AWS Dashboard](https://github.com/elastic/integrations/pull/4596). For that PR, the version was `1.26.0` so I kept version `1.27.0`

**Pull request merged**